### PR TITLE
query cancellation attempt

### PR DESCRIPTION
**Pull request by hrl20**
Implement query cancellation for DuckDB/MotherDuck using conn.interrupt() https://github.com/duckdb/duckdb/pull/7895

### CODE REVIEW

File: dbt/adapters/duckdb/impl.py [hunk 1]
```diff
@@ -51,10 +51,6 @@ class DuckDBAdapter(SQLAdapter):
     def date_function(cls) -> str:
         return "now()"
 
-    @classmethod
-    def is_cancelable(cls) -> bool:
-        return False
```
Comment (jwills) [inline dbt/adapters/duckdb/impl.py:56]: Mmm should we make this environment dependent, possibly?

File: dbt/adapters/duckdb/environments/local.py [hunk 2]
```diff
@@ -58,6 +58,9 @@ def notify_closed(self):
             if self.handle_count == 0 and not self._keep_open:
                 self.close()
 
+    def cancel(cls, connection: Connection):
+        connection.handle.cursor().interrupt()
```
Comment (jwills) [inline dbt/adapters/duckdb/environments/local.py:62]: Is this a new feature of MD or of DuckDB in general?

Comment (hrl20) [reply dbt/adapters/duckdb/environments/local.py:62]: This is a duckdb feature: https://github.com/duckdb/duckdb/pull/7895
I also manually tested it with MD and saw the expected behavior.

**Review by jwills — APPROVED** [review]
Looks great @hrl20, thank you so much!

**Comment by jwills** [comment]
The failing tests are the flaky attach test-- I'm writing a PR to make it non-flaky

**Pull request merged**